common/collision.go

```diff
@@ -37,33 +40,84 @@ func (sc SpaceComponent) AABB() engo.AABB {
 		}
 	}
 
-	sin, cos := math.Sincos(sc.Rotation * math.Pi / 180)
-	xmin := sc.Position.X
-	xmax := sc.Position.X + sc.Width*cos - sc.Height*sin
-	ymin := sc.Position.Y
-	ymax := sc.Position.Y + sc.Height*cos + sc.Width*sin
+	corners := sc.Corners()
 
 	var (
-		X_MIN, X_MAX, Y_MIN, Y_MAX float32
+		X_MIN float32 = -math.MaxFloat32
+		X_MAX float32 = math.MaxFloat32
+		Y_MIN float32 = -math.MaxFloat32
+		Y_MAX float32 = math.MaxFloat32
 	)
 
-	if xmin < xmax {
-		X_MIN = xmin
-		X_MAX = xmax
-	} else {
-		X_MIN = xmax
-		X_MAX = xmin
+	for i := 0; i < 4; i++ {
+		if corners[i].X < X_MIN {
+			X_MIN = corners[i].X
+		} else if corners[i].X > X_MAX {
+			X_MAX = corners[i].X
+		}
+		if corners[i].Y < Y_MIN {
+			Y_MIN = corners[i].Y
+		}
+		if corners[i].Y > Y_MAX {
+			Y_MAX = corners[i].Y
+		}
 	}
 
-	if ymin < ymax {
-		Y_MIN = ymin
-		Y_MAX = ymax
-	} else {
-		Y_MIN = ymax
-		Y_MAX = ymin
+	return engo.AABB{engo.Point{X_MIN, Y_MIN}, engo.Point{X_MAX, Y_MAX}}
+}
+
+// Corners returns the location of the four corners of the rectangular plane defined by the `SpaceComponent`, taking
+// into account any possible rotation.
+func (sc SpaceComponent) Corners() (points [4]engo.Point) {
+	points[0].X = sc.Position.X
+	points[0].Y = sc.Position.Y
+
+	sin, cos := math.Sincos(sc.Rotation * math.Pi / 180)
```

Comment:
You would think that `math.Sincos` can be skipped if `Rotation == 0`, and that is true.

However, reliably checking whether `Rotation == 0`, requires additional resources which make it not worth it.